Adding wiring pi

bseto opened the pull request.

bseto committed:
Added preprocessor if statements

bseto committed:
First Commit with WiringPi files

bseto committed:
Testing inits on wiringPi

bseto committed:
Testing the function on pi

bseto committed:
Changed scheme to work on PullUp

bseto committed:
Init all buttons

bseto committed:
Added breaks to the switch

bseto committed:
Adding timing

bseto committed:
Adding namespace

bseto committed:
Switching songs and changing volumes work

bseto commented:
Do not merge until the debug messages are removed. Will clean the code up on Saturday meeting.

mpminardi commented on SDL/include/WiringPiButtons.hpp:
```diff
@@ -0,0 +1,25 @@
+      int getEvents();  
+
+    private:
+      enum Button { UP = 25, DOWN = 24, LEFT  = 23, RIGHT = 22, RELEASED = 0};
```
Add a comment explaining the reasoning behind these numbers / the mapping to the physical pins.

mpminardi commented on SDL/include/WiringPiButtons.hpp:
```diff
@@ -0,0 +1,25 @@
+    private:
+      enum Button { UP = 25, DOWN = 24, LEFT  = 23, RIGHT = 22, RELEASED = 0};
+      void initButton(int buttonNumber);
+      Button State;
```
Should "State" be capitalized?

mpminardi commented on SDL/include/WiringPiButtons.hpp:
```diff
@@ -0,0 +1,25 @@
+#include <wiringPi.h>
+#include <chrono>
+
+using namespace std::chrono;
```
I do not think we should get in the habit of using namespaces.

mpminardi commented on SDL/src/Main.cpp:
```diff
@@ -176,7 +175,33 @@ void processEvents()
-
+void processGPIO(int button)
+{
+    //enum Button { UP = 25, DOWN = 24, LEFT  = 23, RIGHT = 22, RELEASED = 0};
```
We should be using the Button enum here. Also, remove this commented out code once we do that.

mpminardi commented on SDL/src/WiringPiButtons.cpp:
```diff
@@ -0,0 +1,83 @@
+	        printf("FUNCTION returned %d\n",tempValue);
+                return tempValue; 
+            }
+            if (!digitalRead(UP))       { State = UP; }
```
Make these if statements look like your other if's, or maybe replace them with ternary operators

bseto committed:
Changed Makefile

bseto committed:
Button needed a resolver

bseto committed:
# This is a combination of 2 commits.
# The first commit's message is:

Button needed a resolver

# This is the 2nd commit message:

Changed ints to Button

bseto committed:
Changed button enum names

bseto committed:
Changed names in init

bseto committed:
Changed switch statement to new names

mpminardi commented on SDL/src/Main.cpp:
```diff
@@ -175,7 +166,31 @@ void processEvents()
+	    musicPlayer.changeVolume(-0.1);
+	    break;
+	case WiringPiButtons::TOGGLEPLAY:
+	    printf("Space was pressed!\n");
```
Remove this printf

bseto committed:
Cleaned up main and wiringpi

bseto committed:
Merged conflicts
